users/views.py

```diff
@@ -1,3 +1,42 @@
 from django.shortcuts import render
+from django.views.generic import TemplateView
 
-# Create your views here.
+
+from invoices.models import Invoice, Item
+
+
+class DashboardView(TemplateView):
+    """ Dashboard view for portal details
+    """
+    template_name = 'index.html'
+
+    def get(self, *args, **kwargs):
+        """ Display portal summary details
+        """
+        invoices = Invoice.objects.all()
+
+        count_draft = 0
+        count_due = 0
+
+        for invoice in invoices:
+            amount = 0
+            items = Item.objects.filter(invoice=invoice)
+
+            for item in items: # Get the total amount
```

Comment:
transfer the logic to the models (WHICH SHOULD CONTAIN THE LOGIC)

create mixins.

STEP UP